Yosemite/Yosemite/Tools/POS/POSCart.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+
+/// POSCart is different from the Cart in the POS app layer.
+/// - The POS cart UI might show the cart items differently from how they appear in an order in wp-admin.
+
+public struct POSCart {
+    public let items: [POSCartItem]
+    public let coupons: [POSCartCouponItem]
+
+    public init(items: [POSCartItem] = [], coupons: [POSCartCouponItem] = []) {
+        self.items = items
+        self.coupons = coupons
+    }
+}
+
+public struct POSCartItem {
+    public let item: POSOrderableItem
+    public let quantity: Decimal
+
+    public init(item: POSOrderableItem, quantity: Decimal) {
+        self.item = item
+        self.quantity = quantity
+    }
+}
+
+public struct POSCartCouponItem {
```

Comment:
Do we need this struct when we already have `POSCoupon` with `code`?

Reply:
Good point. I mostly tried to follow the existing patterns for consistency.

As it's mentioned in the comment when `POSCartItem` was created:

> /// POSCart is different from the Cart in the POS app layer.
> /// - The POS cart UI might show the cart items differently from how they appear in an order in wp-admin.

So, for example. `CartCouponItem` may contain `validationStatus` that would show if coupon is valid or invalid but it's a detail only needed for POS UI and not for `Yosemite` layer.

If we never find value in these POS-only entities, we can remove all of them togethen and keep `Cart`, `CartItem`, `CartCouponItem`.

---

That being said, we could use `POSCoupon` instead of `POSCartCouponItem` here.  👍

Reply:
Removed POSCartCouponItem https://github.com/woocommerce/woocommerce-ios/pull/15414/commits/bdf3a9d27a00f80d8a5b5ea29fcafc9fa53a873c

Reply:
> So, for example. CartCouponItem may contain validationStatus that would show if coupon is valid or invalid but it's a detail only needed for POS UI and not for Yosemite layer.

That makes sense 🙇